cishouseholds/impute.py

```diff
@@ -287,70 +288,47 @@ def fill_forward_event(
     visit_id_column
     """
     event_columns = [event_date_column, event_indicator_column, *detail_columns]
-    null_visit_df = df.filter(F.col(visit_datetime_column).isNull())
-    df = df.filter(F.col(visit_datetime_column).isNotNull())
 
-    grouping_window = Window.partitionBy(participant_id_column, event_date_column).orderBy(visit_datetime_column)
-    window = Window.partitionBy(participant_id_column, visit_datetime_column, visit_id_column).orderBy(
-        F.desc(event_date_column)
-    )
-    ordering_window = Window.partitionBy(participant_id_column, event_date_column).orderBy(
-        visit_datetime_column, "DATE_DIFF"
-    )
+    ordering_window = Window.partitionBy(participant_id_column).orderBy(visit_datetime_column)
+    window = Window.partitionBy(participant_id_column, visit_datetime_column, visit_id_column).orderBy("VISIT_DIFF")
```

Comment:
Unsure what visit_diff is?

Reply:
It is necessary to enable disambiguation of the event_dataframe when joining it back onto the main df by visit_datetime.